src/lib.rs

```diff
@@ -1,820 +1,695 @@
-pub type rocksdb_compactionfiltercontext_t = Struct_rocksdb_compactionfiltercontext_t;
-pub enum Struct_rocksdb_compactionfilterfactory_t { }
-pub type rocksdb_compactionfilterfactory_t = Struct_rocksdb_compactionfilterfactory_t;
-pub enum Struct_rocksdb_comparator_t { }
-pub type rocksdb_comparator_t = Struct_rocksdb_comparator_t;
-pub enum Struct_rocksdb_env_t { }
-pub type rocksdb_env_t = Struct_rocksdb_env_t;
-pub enum Struct_rocksdb_fifo_compaction_options_t { }
-pub type rocksdb_fifo_compaction_options_t = Struct_rocksdb_fifo_compaction_options_t;
-pub enum Struct_rocksdb_filelock_t { }
-pub type rocksdb_filelock_t = Struct_rocksdb_filelock_t;
-pub enum Struct_rocksdb_filterpolicy_t { }
-pub type rocksdb_filterpolicy_t = Struct_rocksdb_filterpolicy_t;
-pub enum Struct_rocksdb_flushoptions_t { }
-pub type rocksdb_flushoptions_t = Struct_rocksdb_flushoptions_t;
-pub enum Struct_rocksdb_iterator_t { }
-pub type rocksdb_iterator_t = Struct_rocksdb_iterator_t;
-pub enum Struct_rocksdb_logger_t { }
-pub type rocksdb_logger_t = Struct_rocksdb_logger_t;
-pub enum Struct_rocksdb_mergeoperator_t { }
-pub type rocksdb_mergeoperator_t = Struct_rocksdb_mergeoperator_t;
-pub enum Struct_rocksdb_options_t { }
-pub type rocksdb_options_t = Struct_rocksdb_options_t;
-pub enum Struct_rocksdb_block_based_table_options_t { }
-pub type rocksdb_block_based_table_options_t = Struct_rocksdb_block_based_table_options_t;
-pub enum Struct_rocksdb_cuckoo_table_options_t { }
-pub type rocksdb_cuckoo_table_options_t = Struct_rocksdb_cuckoo_table_options_t;
-pub enum Struct_rocksdb_randomfile_t { }
-pub type rocksdb_randomfile_t = Struct_rocksdb_randomfile_t;
-pub enum Struct_rocksdb_readoptions_t { }
-pub type rocksdb_readoptions_t = Struct_rocksdb_readoptions_t;
-pub enum Struct_rocksdb_seqfile_t { }
-pub type rocksdb_seqfile_t = Struct_rocksdb_seqfile_t;
-pub enum Struct_rocksdb_slicetransform_t { }
-pub type rocksdb_slicetransform_t = Struct_rocksdb_slicetransform_t;
-pub enum Struct_rocksdb_snapshot_t { }
-pub type rocksdb_snapshot_t = Struct_rocksdb_snapshot_t;
-pub enum Struct_rocksdb_writablefile_t { }
-pub type rocksdb_writablefile_t = Struct_rocksdb_writablefile_t;
-pub enum Struct_rocksdb_writebatch_t { }
-pub type rocksdb_writebatch_t = Struct_rocksdb_writebatch_t;
-pub enum Struct_rocksdb_writeoptions_t { }
-pub type rocksdb_writeoptions_t = Struct_rocksdb_writeoptions_t;
-pub enum Struct_rocksdb_universal_compaction_options_t { }
-pub type rocksdb_universal_compaction_options_t = Struct_rocksdb_universal_compaction_options_t;
-pub enum Struct_rocksdb_livefiles_t { }
-pub type rocksdb_livefiles_t = Struct_rocksdb_livefiles_t;
-pub enum Struct_rocksdb_column_family_handle_t { }
-pub type rocksdb_column_family_handle_t = Struct_rocksdb_column_family_handle_t;
-
-pub const rocksdb_block_based_table_index_type_binary_search: c_uint = 0;
-pub const rocksdb_block_based_table_index_type_hash_search: c_uint = 1;
-
-pub const rocksdb_no_compression: c_uint = 0;
```

Comment:
Yes, I reverted to the constants after all. I'm not sure you can say they *exactly* match the C headers, since the C header defines a nameless enum, but there you go! The semantics are similar.